03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,29 +1,106 @@
+  //try {
   const task = await Task.create(req.body);
   res.status(201).json({ task });
-};
+  //} catch (error) {
+  //res.status(500).json({ mesg: error });
+  //}
+});
 
-const getTask = (req, res) => {
+/*const getTask = async (req, res) => {
   //res.send("get single tasks");
-  res.json({ id: req.params.id });
-};
+  //res.json({ id: req.params.id });
+  try {
+    const { id: taskID } = req.params;
+    const task = await Task.findOne({ _id: taskID });
+    if (!task) {
+      return res
+        .status(404)
+        .json({ msg: `not find task with the id:${taskID}` });
+    }
+    res.status(200).json({ task });
+  } catch (error) {
+    res.status(500).json({ mesg: error });
+  }
+};*/
 
-const updateTask = (req, res) => {
-  res.send("update tasks");
-};
+const getTask = asyncWrapper(async (req, res, next) => {
+  const { id: taskID } = req.params;
+  const task = await Task.findOne({ _id: taskID });
+  if (!task) {
+    return next(createCustomError(`not find task with the id:${taskID}`, 404));
+  }
+  res.status(200).json({ task });
+});
 
-const deleteTask = (req, res) => {
-  res.send("delete tasks");
-};
+/*const deleteTask = async (req, res) => {
+  // res.send("delete tasks");
+  try {
+    const { id: taskID } = req.params;
+    const task = await Task.findOneAndDelete({ _id: taskID });
+    if (!task) {
+      return res
+        .status(404)
+        .json({ msg: `not find task with the id:${taskID}` });
+    }
+    //res.status(200).json({ task }); una forma de escribirlo
+    //res.status(200).send() otra forma de escribirlo
+    res.status(200).json({ task: null, status: "success" }); //otra forma de escribirlo
+  } catch (error) {
+    res.status(500).json({ mesg: error });
+  }
+};*/
+
+const deleteTask = asyncWrapper(async (req, res) => {
+  const { id: taskID } = req.params;
+  const task = await Task.findOneAndDelete({ _id: taskID });
+  if (!task) {
+    //return res.status(404).json({ msg: `not find task with the id:${taskID}` });
+    return next(createCustomError(`not find task with the id:${taskID}`, 404));
+  }
+  res.status(200).json({ task: null, status: "success" }); //otra forma de escribirlo
+});
+
+/*const updateTask = async (req, res) => {
+  //res.send("update tasks");
+  try {
+    const { id: taskID } = req.params;
+    // res.status(200).json({ id: taskID, data: req.body });
+    const task = await Task.findOneAndUpdate({ _id: taskID }, req.body, {
+      new: true,
+      runValidators: true,
+    });
+  } catch (error) {
+    res.status(500).json({ mesg: error });
+  }
+};*/
+
+const updateTask = asyncWrapper(async (req, res) => {
+  const { id: taskID } = req.params;
+  const task = await Task.findOneAndUpdate({ _id: taskID }, req.body, {
+    new: true,
+    runValidators: true,
+  });
+
+  if (!task) {
+    //return res.status(404).json({ msg: `not task with id: ${taskID}` });
+    return next(createCustomError(`not find task with the id:${taskID}`, 404));
```

Comment:
know the video doesn't include adding `next` in the method signature above `(req, res, next)`, but for clarity it's good to have it there